### PR TITLE
Skeleton Additon

### PR DESCRIPTION
**Pull request by Kelzihar**
This is my addition of the skeleton for the rest of the idea I had. I also added a function to check the line's indentation called indentationCheck. Please provide any feedback you may have or ask questions in teams if there are any. Otherwise if everything looks good approve and merge.

### CODE REVIEW

**Comment by Jsatram** [comment]
👍

**Review by Jsatram — APPROVED** [review]

File: syntax-analysis.py [hunk 1]
```diff
@@ -12,39 +20,70 @@ def fileAnalysis(file):
...
                         line = line[:line.find('def') + 3] + '(' + line[line.find('def') + 3:]
                         out.write(line)
-                        opened.readline()
+                        line = opened.readline()
                     elif line.find(')') == -1:
                         line = line[:line.find(':')] + ')' + line[line.find(':'):]
                         out.write(line)
-                        opened.readline()
+                        line = opened.readline()
                     else:
```
Comment (Jsatram) [inline syntax-analysis.py]: if the line is missing both '(' and ')' then this else case will never be met and the first condition will be met instead.

File: syntax-analysis.py [hunk 2]
```diff
@@ -12,39 +20,70 @@ def fileAnalysis(file):
...
+                    line, sub = indentationCheck(line, sub) # Example using new indentation check method and skeleton for rest
+                    out.write(line)
+                    line = opened.readline()
+                elif line.find('(') == -1: 
+                    line = line[:line.find('def') + 3] + '(' + line[line.find('def') + 3:]
+                    # Input measures for the indentation check
+                elif line.find(')') == -1:
+                    line = line[:line.find(':')] + ')' + line[line.find(':'):]
+                    # Input measures for indentation check
+                else:
```
Comment (Jsatram) [inline syntax-analysis.py]: the same comment mentioned earlier, this case will never be met.
Check to make sure only one element is missing at a time so the else case is met correctly.

**Comment by Kelzihar** [comment]
I have just gone through and reworked that section. Switched from elif to if, since really all of those values should be checked every time.

**Pull request merged**